### PR TITLE
[UI][FIX] Disabled Signup button for the users who already have the extension/the capability to use meshmap

### LINKED ISSUE #7155: [UI] Extensions: Enhance logic to include user's/provider's capabilities
#### Current Behavior
Whether users have access to a particular extension or not, they are shown the same message to "sign up".

Screenshot 2023-02-22 at 9 23 05 AM

#### Desired Behavior
If the remote provider for the currently logged in user offers the extension/the capability, (if the user has already signed up and has access), then the "Sign Up" button should be disabled.

#### Mockups

---

#### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)

- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by krishna9304**
**Notes for Reviewers**

This PR fixes #7155

**Approach:**
- Fetched the `capabiltiesRegistry` from the global store.
- Checked whether there exists an extension navigator for `meshmap` for the currently signed in user.
- Finally disabling the signup button and the signup message.

## Demo

https://user-images.githubusercontent.com/71918441/221029801-cb86f4ad-c704-4731-9881-58782cd8b20b.mov

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [X] Yes, I signed my commits.

cc: @leecalcote

### CODE REVIEW

**Comment by leecalcote** [comment]
This is looking real good. @krishna9304, while disabling the card on the Extensions page, will you also change the button text from "Sign Up" to "Enabled"?

**Comment by leecalcote** [comment]
@MUzairS15 any thoughts on this one?

File: ui/pages/extensions.js [hunk 1]
```diff
@@ -152,14 +154,22 @@ const Extensions = ({ classes, toggleCatalogContent,  enqueueSnackbar, closeSnac
     )
   }
 
+  useEffect(() => {
+    const temp = capabilitiesRegistry?.extensions?.navigator?.filter(
```
Comment (MUzairS15) [inline ui/pages/extensions.js:158]: Try giving meaningful names.

File: ui/pages/extensions.js [hunk 2]
```diff
@@ -201,11 +211,16 @@ const Extensions = ({ classes, toggleCatalogContent,  enqueueSnackbar, closeSnac
...
-  catalogVisibility : state.get('catalogVisibility')
+  catalogVisibility : state.get('catalogVisibility'),
+  capabilitiesRegistry : state.get('capabilitiesRegistry')
 })
 
 const mapDispatchToProps = dispatch => ({
   toggleCatalogContent : bindActionCreators(toggleCatalogContent, dispatch)
 })
 
+export const extensions = {
```
Comment (MUzairS15) [inline ui/pages/extensions.js:222]: Define this constant in Enum.js (where all other constants are defined)

File: ui/pages/extensions.js [hunk 3]
```diff
@@ -201,11 +211,16 @@ const Extensions = ({ classes, toggleCatalogContent,  enqueueSnackbar, closeSnac
...
+  meshmap : "MeshMap"
```
Comment (MUzairS15) [inline ui/pages/extensions.js:223]:
```suggestion
  MESHMAP : "meshmap"
```
So that you don't need to do `.toLowerCase()`

**Review by MUzairS15 — CHANGES_REQUESTED** [review]
Looks good, just some minor things to consider.

**Comment by krishna9304** [comment]
> This is looking real good. @krishna9304, while disabling the card on the Extensions page, will you also change the button text from "Sign Up" to "Enabled"?

Sure, Will do that.

File: ui/pages/extensions.js (hunk 1 above)
Comment (krishna9304) [reply ui/pages/extensions.js:158]: Sure.

File: ui/pages/extensions.js (hunk 2 above)
Comment (krishna9304) [reply ui/pages/extensions.js:222]: okay.

File: ui/pages/extensions.js (hunk 3 above)
Comment (krishna9304) [reply ui/pages/extensions.js:223]: Noted! I will do that.

**Review by leecalcote — APPROVED** [review]
LGTM. 

@theBeginner86?

**Review by theBeginner86 — APPROVED** [review]

**Comment by theBeginner86** [comment]
Thank you @krishna9304 🚀

**Pull request merged**